upstairs/src/downstairs.rs

```diff
@@ -1079,6 +1115,20 @@ impl Downstairs {
             state,
         });
 
+        info!(
+            self.log,
+            "starting repair {}",
+            self.repair.as_ref().unwrap().id
+        );
+
+        #[cfg(feature = "omicron-build")]
+        {
+            // If building for production, then notify Nexus when any
+            // live repair starts.
+            let cloned_repair = self.repair.as_ref().unwrap().clone();
```

Comment:
I don't love deriving `Clone` on `LiveRepairData`, because it contains a bunch things that are pretty expensive (`BTreeMap`, `Vec`, etc).

It looks like we can make it work with borrows:
```diff
diff --git a/upstairs/src/downstairs.rs b/upstairs/src/downstairs.rs
index 0c79137..5d313f0 100644
--- a/upstairs/src/downstairs.rs
+++ b/upstairs/src/downstairs.rs
@@ -160,7 +160,7 @@ impl LiveRepairState {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub(crate) struct LiveRepairData {
     /// An ID uniquely identifying this repair
     id: Uuid,
@@ -1125,8 +1125,8 @@ impl Downstairs {
         {
             // If building for production, then notify Nexus when any
             // live repair starts.
-            let cloned_repair = self.repair.as_ref().unwrap().clone();
-            self.notify_nexus_of_live_repair_start(cloned_repair);
+            let repair = self.repair.as_ref().unwrap();
+            self.notify_nexus_of_live_repair_start(repair);
         }
 
         // We'll be back in on_live_repair once the initial job finishes
@@ -1400,8 +1400,8 @@ impl Downstairs {
                 {
                     // If building for production, then notify Nexus when any
                     // live repair completes.
-                    let cloned_repair = repair.clone();
-                    self.notify_nexus_of_live_repair_finish(cloned_repair);
+                    let repair = self.repair.as_ref().unwrap();
+                    self.notify_nexus_of_live_repair_finish(repair);
                 }
 
                 // Set `self.repair` to `None` on our way out the door (because
@@ -3823,7 +3823,7 @@ impl Downstairs {
     }
 
     #[cfg(feature = "omicron-build")]
-    fn get_target_addrs(&mut self) -> Vec<SocketAddr> {
+    fn get_target_addrs(&self) -> Vec<SocketAddr> {
         self.clients
             .iter()
             .map(|client| client.target_addr.clone())
@@ -3832,7 +3832,7 @@ impl Downstairs {
     }
 
     #[cfg(feature = "omicron-build")]
-    fn notify_nexus_of_live_repair_start(&mut self, repair: LiveRepairData) {
+    fn notify_nexus_of_live_repair_start(&self, repair: &LiveRepairData) {
         use crate::get_nexus_client;
         use chrono::Utc;
         use nexus_client::types::DownstairsUnderRepair;
@@ -3915,7 +3915,7 @@ impl Downstairs {
     }
 
     #[cfg(feature = "omicron-build")]
-    fn notify_nexus_of_live_repair_finish(&mut self, repair: LiveRepairData) {
+    fn notify_nexus_of_live_repair_finish(&self, repair: &LiveRepairData) {
         use crate::get_nexus_client;
         use chrono::Utc;
         use nexus_client::types::DownstairsUnderRepair;
```

(the change to `get_target_addrs` is just to make it take `&self`, since it's not actually mutating the `Downstairs`)

Reply:
nice, done in 3c0ca0d